pytrickle/client.py

```diff
@@ -108,6 +112,18 @@ async def start(self, request_id: str = "default"):
         finally:
             self.running = False
             logger.info("Stopping protocol due to client loops ending")
+            
+            # Call the optional on_stream_stop callback before stopping protocol
+            if hasattr(self.frame_processor, 'on_stream_stop') and self.frame_processor.on_stream_stop:
+                try:
+                    if asyncio.iscoroutinefunction(self.frame_processor.on_stream_stop):
```

Comment:
In another PR moving load_model and param_updates to async, thinking we just keep all async? WDYT?

Reply:
Yes, the option should be async. I rebased the async changes from main in https://github.com/livepeer/pytrickle/pull/23

Reply:
Updated the error_callback and on_stream_stop callbacks to be async https://github.com/livepeer/pytrickle/pull/25/commits/b16eb1f72c6f7519662b9b6a2275ea0fbd9d060e